src/devices/SocketCan/Interop.cs

```diff
@@ -30,106 +29,54 @@ internal class Interop
         public const int SOL_CAN_BASE = 100;
         public const int SOL_CAN_RAW = SOL_CAN_BASE + (int)CanProtocol.CAN_RAW;
 
-        [DllImport("libc", EntryPoint = "socket", CallingConvention = CallingConvention.Cdecl)]
-        private static extern int CreateNativeSocket(int domain, int type, CanProtocol protocol);
-
         [DllImport("libc", EntryPoint = "ioctl", CallingConvention = CallingConvention.Cdecl)]
         private static extern int Ioctl3(int fd, uint request, ref ifreq ifr);
 
-        [DllImport("libc", EntryPoint = "bind", CallingConvention = CallingConvention.Cdecl)]
-        private static extern int BindSocket(int fd, ref CanSocketAddress addr, uint addrlen);
-
-        [DllImport("libc", EntryPoint = "close", CallingConvention = CallingConvention.Cdecl)]
-        private static extern int CloseSocket(int fd);
-
-        [DllImport("libc", EntryPoint = "write", CallingConvention = CallingConvention.Cdecl)]
-        private static unsafe extern int SocketWrite(int fd, byte* buffer, int size);
-
-        [DllImport("libc", EntryPoint = "read", CallingConvention = CallingConvention.Cdecl)]
-        private static unsafe extern int SocketRead(int fd, byte* buffer, int size);
-
         [DllImport("libc", EntryPoint = "setsockopt", CallingConvention = CallingConvention.Cdecl)]
         private static unsafe extern int SetSocketOpt(int fd, int level, int optName, byte* optVal, int optlen);
 
```

Comment:
As much as I wish that these `extern`s were gone, we can't use `Socket.IOControl` (it is almost Windows-only) and `Socket.SetSocketOption` (in `SocketOptionLevel` there's no definition for CAN bus, and PAL verifies provided values, so we cannot write them manually).

Reply:
can we convert rest of the code but leave this one? That still should be much better than re-implementing socket 😄